src/components/App.jsx

```diff
@@ -87,14 +89,15 @@ class App extends Component {
 
     render() {
         const {
-            loading,
             metadata,
             error,
             loaded,
             isSidebarVisible,
             sessionsLimitReached,
         } = this.state;
 
+        const { isLoading } = this.props;
```

Comment:
I expected the `Component` to be parameterised by a custom prop type definition?